include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -97,41 +97,140 @@ template <typename> class FullMatrix;
  * @author Martin Kronbichler, 2017
  */
 template <int dim, typename Number, int size = -1>
+class TensorProductMatrixSymmetricSumBase
+{
+public:
+  /**
+   * Returns the number of rows of this matrix, given by the dim-th power of
+   * the size of the 1D matrices passed to the constructor.
+   */
+  unsigned int m () const;
+
+  /**
+   * Returns the number of columns of this matrix, given by the dim-th power
+   * of the size of the 1D matrices passed to the constructor.
+   */
+  unsigned int n () const;
+
+  /**
+   * Implements a matrix-vector product with the underlying matrix as
+   * described in the main documentation of this class. Same as the other
+   * vmult() function, but operating on plain pointers rather than a vector
+   * (no check of array bounds possible).
```

Comment:
You already noted that you will need to update the documentation: Here is a spot where you need to avoid to refer to the `vmult()` function that now sits in the derived class. Copy that documentation here instead.

Reply:
Okay, I will take care of it. By the way, shall the main documentation stay with the base class or is it better to add this to the derived classes?

Reply:
I would say we should list it with the derived class that a user sees. Since we have one class and a specialization, I suggest we put all documentation to the generic type with `Number` argument and use something like
```
/** @copydoc TensorProductMatrixSymmetricSum<Number>
* This is the specialization for `VectorizedArray`...
*/
```